addons/account/models/account_tax.py

```diff
@@ -2382,14 +2393,17 @@ def _prepare_tax_lines(self, base_lines, company, tax_lines=None):
                     tax_line['tax_base_amount'] += sign * tax_data['base_amount'] * (-1 if tax_tag_invert else 1)
                     tax_line['amount_currency'] += sign * tax_rep_data['tax_amount_currency']
                     tax_line['balance'] += sign * tax_rep_data['tax_amount']
+                    tax_line['keep_zero_line'] |= self._should_keep_zero_line(base_line, tax_data, tax_rep_data)
```

Comment:
put this on `_prepare_base_line_tax_repartition_grouping_key`

Reply:
Oh sorry, I see now what you meant. That would also split the lines appropriately if `keep_zero_line` was different (although again, for external taxes it won't differ in the same record).

I do have to reconstruct all the `frozendict`s now to remove `keep_zero_line`, because it's not a field on `account.move.line`. Not ideal, but I can't immediately see a better way. Filtering it out in `create/write` or similar is too hacky I think. Perhaps I should do the filtering in a method so it's more extendable? Let me know your preference!